### PR TITLE
Refactor before moving forward with core game logic

### LINKED ISSUE #5: Rename Handlers & Composers
Make the packet handlers/composers have naming that is more aligned with the Habbo client source  code

### PR DESCRIPTION
**Pull request by jtieri**
- [x] fix config stuff
- [x] switch to postgres
- [x] make logging sane and consistent
- [x] think about removing singleton app context
- [x] cleanup code all over
- [x] refactor protocol package to be reflective of the habbo src

Closes #5

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 import (
+	"fmt"
 	"gopkg.in/yaml.v2"
 	"io/ioutil"
 	"log"
 	"os"
+	"path"
 )
 
+var defaultHome = os.ExpandEnv("$HOME/.habbgo")
```
Comment (emansom) [inline config/config.go:12]: https://github.com/adrg/xdg
https://pkg.go.dev/github.com/adrg/xdg#ConfigFile

Comment (jtieri) [reply config/config.go:12]: Ohh snap! This is awesome, gonna throw it in here right now.

Comment (jtieri) [reply config/config.go]: went ahead and added it in. this was a great addition, thanks for the tip!

**Comment by jtieri** [comment]
Gonna try to get standardized logging in place across the current codebase. After that I'll do one more audit of the project structure/design and cleanup then should be good to move forward with implementing the game.

**Pull request merged**